crates/p2p/src/client/types.rs

```diff
@@ -254,7 +256,51 @@ impl TryFromDto<p2p_proto::transaction::Transaction> for TransactionVariant {
                     .collect(),
                 compiled_class_hash: CasmHash(x.compiled_class_hash),
             }),
-            DeclareV3(_) => unimplemented!(),
+            DeclareV3(x) => TransactionVariant::DeclareV3(DeclareTransactionV3 {
+                class_hash: ClassHash(x.class_hash.0),
+                nonce: TransactionNonce(x.nonce),
+                nonce_data_availability_mode:
+                    pathfinder_common::transaction::DataAvailabilityMode::L1,
+                fee_data_availability_mode:
+                    pathfinder_common::transaction::DataAvailabilityMode::L1,
```

Comment:
Do these fields just not exist yet?

Reply:
They do -- this was just a shortcut. I've added proper TryFromDto implementation for DataAvailabilityMode and use that here.